File: MyProjects/MAP/Assignment 6/src/Model/Expressions/ConstExpr.java
package Model.Expressions;

import Model.Utilities.IDictionary;

public class ConstExpr implements IExpr {
    private int number;

    public ConstExpr(int nr){
        number=nr;
    }

    @Override
    public int evaluate(IDictionary<String, Integer> dic, IDictionary<Integer, Integer> heap) {
        return number;
    }

    public String toString(){
        return "" + number;
    }
}

File: MyProjects/MAP/Assignment 7/Interpreter/Model/Expressions/ConstExpr.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Interpreter.Model.Expressions
{
    public class ConstExpr:IExpr
    {
        private int number;

        public ConstExpr(int nr)
        {
            number = nr;
        }

        public int evaluate(Utilities.IDictionary<string, int> dic)
        {
            return number;
        }

        public override string ToString()
        {
            return "" + number;
        }
    }
}
